project/Classroom.cs

```diff
@@ -2,13 +2,50 @@ namespace assignment_3;
 
 public class Classroom
 {
-    public int RoomId { get; set; }
-    public int RoomNumber { get; set; }
-    public int Capacity { get; set; }
+    private static List<Classroom> _classrooms = new List<Classroom>();
 
+    private int _roomId;
+    private int _capacity;
+
+    public int RoomId
+    {
+        get => _roomId;
+        private set
+        {
+            if(value<=0) throw new ArgumentException("Room id cannot be less than 0");
+            _roomId = value;
+        }
+    }
+    public int Capacity
+    {
+        get => _capacity; 
+        set
+        {
+            if (value < 0) throw new ArgumentException("Capacity cannot be negative.");
+            _capacity = value;
+        }
+    }
     public bool CheckAvailability()
     {
         return Capacity > 0;
+
+    }
+    public Classroom(int roomId, int capacity)
+    {
+        RoomId = roomId;  // invoke the setter with validation
+        Capacity = capacity; 
+        AddClassroom(this);  //instance add
         
     }
+
+    public static List<Classroom> GetClassrooms()
```

Comment:
We don't return the reference of the list, we return a copy. So just:

return new List<Classroom>(_classrooms_List);

Also must be moved to the ClassExtents file. This is all addressed in my PR.